src/pykeen/regularizers.py

```diff
@@ -70,7 +67,7 @@ def get_normalized_name(cls) -> str:
 
     def reset(self) -> None:
         """Reset the regularization term to zero."""
-        self.regularization_term = torch.zeros(1, dtype=torch.float, device=self.device)
+        self.regularization_term = torch.zeros(1, dtype=torch.float)
```

Comment:
Here, I am not sure whether it still stays a buffer, it we assign a new tensor to it. 

An option would be the fill the tensor in-place with zeros and detach?
```python
self.regularization_term.detach_().zeros_()
```

we should check (via the training unittest), whether this breaks gradient computation.

Reply:
This snipped didn't work, so i switched to 
```python
torch.zeros(size=self.regularization_term.shape, out=self.regularization_term)
```

which gave:
```python-traceback
Error
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.6_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/local/Cellar/python@3.8/3.8.6_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/local/Cellar/python@3.8/3.8.6_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/Users/cthoyt/dev/pykeen/tests/cases.py", line 626, in test_model
    model.post_parameter_update()
  File "/Users/cthoyt/dev/pykeen/src/pykeen/models/base.py", line 850, in post_parameter_update
    super().post_parameter_update()
  File "/Users/cthoyt/dev/pykeen/src/pykeen/models/base.py", line 598, in post_parameter_update
    self.regularizer.reset()
  File "/Users/cthoyt/dev/pykeen/src/pykeen/regularizers.py", line 66, in reset
    torch.zeros(size=self.regularization_term.shape, out=self.regularization_term)
RuntimeError: cannot resize variables that require grad
```

Reply:
`self.regularization_term.detach_().zero_()` works!